candies/cli/cli.py

```diff
@@ -57,6 +61,60 @@ def help(self, *args, **kwargs):
         """
         return self.main.help(*args, **kwargs)
 
+    def parse(self, callable: Union[Callable[[str], Any], Type]) \
+            -> Union[Type, Callable[[str], Any]]:
+        """Defines a parser for a custom type.
+
+        Args:
+            callable: A callable that converts a string to an instance
+                of a custom type. For example, a parser function or a
+                type that accepts a string to a constructor.
+
+        Examples:
+            @cli
+            def add(money: Arg[Money]):
+                ...
+
+            @main.parse
+            def money(x: str) -> Money:
+                if x[0] == '$':
+                    return Money(float(x[1:]), 'dollars')
+                ...
+
+            # Or, as an alternative, if
+            # `Money.__init__(str)` can be called.
+            main.parse(Money)
+        """
+
+        if isinstance(callable, type):
```

Comment:
If we allow passing any kind of objects here, seems like we need to handle this case: `main.parse(5)`.

Reply:
What do you mean when you say "allowing"?

Reply:
I meant that even considering the fact that `parse` accepts an argument of type `Union[Callable[[str], Any]` it's still allowed for user to call it like `parse(5)` or `parse("x")`. So I was wondering, what kind of exception would be thrown in that case.

_3 hours later._

Well, actually, I've checked what kind of exception the `inspect.signature` could throw, and it seems like `TypeError: {x} is not callable object` fits pretty well. Didn't know that, sorry. I was just worried about the fact that `inspect.signature` could throw a too specific error.

Reply:
Oh, I guess I now see what you meant! So, is it okay to leave as it is?

Reply:
Yep.